File: src/main/java/io/tatum/model/request/CustomerUpdate.java
package io.tatum.model.request;

import io.tatum.model.response.ledger.Fiat;
import lombok.*;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;

/**
 * The type Customer update.
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@EqualsAndHashCode
public class CustomerUpdate {

    private Country customerCountry;

    private Fiat accountingCurrency;

    private Country providerCountry;

    @Size(min = 1, max = 100)
    @NotEmpty
    private String externalId;

}

File: src/main/kotlin/io/tatum/model/request/CustomerUpdate.kt
package io.tatum.model.request

import io.tatum.model.response.ledger.Fiat

class CustomerUpdate(
    val customerCountry: Country? = null,
    val accountingCurrency: Fiat? = null,
    val providerCountry: Country? = null,
    val externalId: String,
) {
    init {
        require(externalId.length in 1..100) {"Length of externalId has to be in range 1..100."}
    }
}